lisa/sut_orchestrator/libvirt/platform.py

```diff
@@ -567,6 +570,25 @@ def _create_node(
                         Path(node_context.os_disk_base_file_path),
                     )
 
+                    runbook: BaseLibvirtNodeSchema = environment.runbook
+                    os_disk_size_in_gb = runbook.disk_size_gb
+
+                    if os_disk_size_in_gb:
+                        qemu_tool = self.host_node.tools[QemuImg]
+                        src_file = node_context.os_disk_base_file_path
+                        qemu_tool.resize(
```

Comment:
It would probably be better to move this logic into `_create_node_os_disk` to keep the original OS disk file unchanged.

Also, I believe `qemu-img resize` works on qcow2 diff disks.

Reply:
Done

Reply:
@cwize1 Moving the logic to `_create_node_os_disk ` , for qemu case, will also moodift the original disk file since we use qcow diffs on `os_disk_base_file_path` itself.  https://github.com/microsoft/lisa/blob/main/lisa/sut_orchestrator/libvirt/qemu_platform.py#L42

Having it here right after scp will avoid multiple resize operations.

Reply:
The code was very explicitly designed to avoid modifying the original file, to ensure tests are repeatable.

Also, for qcow2 files, resize operations are very quick. (I don't know about raw files. But I suspect it is also pretty quick.)

Reply:
Moved resize to os_disk_file_path.